lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1580,7 +1580,8 @@ def customize_vm(self, vm_obj):
             # Setting hostName, orgName and fullName is mandatory, so we set some default when missing
             ident.userData.computerName = vim.vm.customization.FixedName()
             # computer name will be truncated to 15 characters if using VM name
-            default_name = self.params['name'].translate(None, string.punctuation)
+            default_name = self.params['name'].replace(' ', '')
+            default_name = ''.join([c for c in default_name if c not in string.punctuation])
```

Comment:
Just a note that if you are in an environment where all strings should be text strings (controller-only code), there are several better ways to do this:

``` python
# If you can require Python-3 only
table = str.maketrans('', '', string.punctuation)
table[ord(' ')] = None
default_name = self.params['name'].translate(table)

# If you can require Python-2.7+
table = {ord(c): None for c in string.punctuation}
table[ord(' ')] = None
default_name = self.params['name'].translate(table)

# otherwise:
from ansible.module_utils.six.moves import zip_longest
table = dict(zip_longest((ord(c) for c in string.punctuation), None))
table[ord(' ')] = None
default_name = self.params['name'].translate(table)
```

Just something to keep in your toolbox for the day when you'll eventually be able to use it :-)

Reply:
Nice, I will keep a note of this for future.